weather_mv/loader_pipeline/bq.py

```diff
@@ -174,6 +194,49 @@ def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.Lis
                             region=pipeline_options_dict.get('region'))
             logger.info('Region validation completed successfully.')
 
+    def generate_csv(
+        self,
+        csv_path: str,
+        lats: t.List,
+        lons: t.List,
+        lat_grid_resolution: float,
+        lon_grid_resolution: float,
+        skip_creating_polygon: bool = False,
+    ):
+        """Generates geo data CSV."""
+        logger.info("Generating geo data CSV ...")
+        # Generate Cartesian product of latitudes and longitudes.
+        lat_lon_pairs = itertools.product(lats, lons)
+        # Create a temp CSV file for writing.
+        with tempfile.NamedTemporaryFile(suffix='.csv', mode='w+', newline='') as temp:
+            # Define header.
+            header = ['latitude', 'longitude', GEO_POINT_COLUMN, GEO_POLYGON_COLUMN]
+            data = []
+            for lat, lon in lat_lon_pairs:
+                lat = float(lat)
+                lon = float(lon)
+                row = [lat, lon]
+                sanitized_lon = (((lon % 360) + 540) % 360) - 180
+
+                # Fetch the geo point.
+                geo_point = fetch_geo_point(lat, sanitized_lon)
+                row.append(geo_point)
+
+                # Fetch the geo polygon if not skipped.
+                if not skip_creating_polygon:
+                    geo_polygon = fetch_geo_polygon(lat, sanitized_lon, lat_grid_resolution, lon_grid_resolution)
+                    row.append(geo_polygon)
+                else:
+                    row.append(None)
+                data.append(row)
+
+            df = pd.DataFrame(data, columns=header)
+            # Write DataFrame to CSV.
+            df.to_csv(temp.name, index=False)
```

Comment:
If this was parquet, for example, you could directly write to the bucket.

Reply:
I benchmarked this, and it looks like the current approach is faster! :)